README.md

```diff
@@ -94,76 +89,128 @@ our [official documentation](https://docs.conjur.org/Latest/en/Content/Developer
-store is not one we support or is not accessible, the credentials will be written to a configuration
-file, `.netrc`, in plaintext.
+#### Step 3. Storing credentials
 
-If written to the `.netrc`, it is strongly recommended that you log out when not using the Conjur
-CLI. This removes the credentials from the `.netrc` file.
+The client retrieves credentials from a credential store called `CredentialStore` which inherits from `CredentialsStoreInterface`. This
+approach enables storing the credentials in a safe location, and providing the credentials to the client on demand.
 
-The `.netrc` file or (`_netrc` for Windows environments) contains credentials needed to log in to
-the Conjur endpoint and should consist of 'machine', 'login', and 'password'.
+The SDK provides the following implementations of `CredentialsStoreInterface`:
+-  `KeystoreCredentialsProvider` which saves credentials to the system's credential store (macOS keychain for example)
+-  `FileCredentialsProvider` which saves the credentials into `.netrc` file, in plaintext.
 
-Note that if you choose to create this file yourself, ensure you follow least privilege, allowing
-only the user who has created the file to have read/write permissions on it (`chmod 700 .netrc`).
+
+We provide the`CredentialStoreFactory` which creates such credential stores.
+
+By default, the `CredentialStoreFactory` favors creating a `KeystoreCredentialsProvider`. If the SDK cannot access the 
+operating system's credential store, then `CredentialStoreFactory` will create `FileCredentialsProvider` instead.
+
+If credentials are written to the `.netrc`, we strongly recommend deleting those credentials when not using the
+SDK. The `.netrc` file is located in the user's home directory, for example: `/Users/my_username` in macOS 
+or `C:\Users\my_username` in Windows
+
+The `.netrc` file or (`_netrc` for Windows environments) contains credentials needed to log in to the Conjur server
+and should consist of 'machine', 'login', and 'password'.
 
 ```
 # .netrc / _netrc
-machine https://conjur.myorg.com
+machine https://conjur-myorg.example.com
 login admin
 password 1234....
 ```
 
-## Supported Client methods
+Note: If you choose to create this file yourself, make sure to follow least privilege rule, allowing only the user who has
+created the file to have read/write permissions on it (`chmod 600 .netrc`).
+
+
+Example of usage:
+
+- Create credential store:
+  ```
+  credentials_store = CredentialStoreFactory.create_credential_store()
+  ```
+
+- Store credentials inside the store
+  ```
+  credentials = CredentialsData(login=username, password=api_key, machine=conjur_url)
+  credentials_store.save(credentials)
+  del credentials # After being saved, it's consider best practice to delete credentials from memory after usage
+  ```
+  - Note: This step is needed only once! Credential stores provided by the SDK are meant to be persistent and can 
```

Comment:
required instead of needed

Reply:
How about "Note: You need to do this only once."